src/awkward/forms/numpyform.py

```diff
@@ -206,3 +207,10 @@ def _select_columns(self, index, specifier, matches, output):
 
     def _column_types(self):
         return (ak.types.numpytype.primitive_to_dtype(self._primitive),)
+
+    def _smallest_zero_buffer_lengths(self) -> Iterator[ShapeItem]:
+        if len(self.inner_shape) > 0:
+            raise NotImplementedError
```

Comment:
Just like `ak.forms.regularform._smallest_zero_buffer_lengths`, this can multiply all of the `inner_shape` items except the last (unless I'm off-by-one).

Reply:
Oops, that was a todo!